homeassistant/components/onvif/__init__.py

```diff
@@ -155,3 +158,32 @@ async def async_populate_options(hass: HomeAssistant, entry: ConfigEntry) -> Non
     }
 
     hass.config_entries.async_update_entry(entry, options=options)
+
+
+@callback
+def _migrate_camera_entities_unique_ids(
+    hass: HomeAssistant, config_entry: ConfigEntry, device: ONVIFDevice
+) -> None:
+    """Migrate unique ids of camera entities from profile index to profile token."""
+    entity_reg = er.async_get(hass)
+    entities: list[er.RegistryEntry] = er.async_entries_for_config_entry(
+        entity_reg, config_entry.entry_id
+    )
+    mac_or_serial = device.info.mac or device.info.serial_number
+    for entity in entities:
+        if entity.domain == "camera":
```

Comment:
nit: I prefer return early pattern to reduce nesting, e.g.

```
if entity.domain != "camera":
  continue
if not entity.unique_id.startswith(old_uid_start) and entity.unique_id != mac_or_serial:
  continue
```

which helps with readability